Nautilus/Assets/ModPrefabCache.cs

```diff
@@ -1,10 +1,74 @@
+using System.Collections.Generic;
+using UnityEngine;
+
 namespace Nautilus.Assets;
 
 /// <summary>
-/// Class used by <see cref="PrefabInfo"/> to store game objects that used as prefabs.
-/// Game objects in cache are inactive and will not be on scene.
+/// Class used by the prefab system to store GameObjects.
+/// Objects in the cache are inactive because they are placed within an inactive parent object.
 /// </summary>
 public static class ModPrefabCache
 {
-    
+    private static ModPrefabCacheInstance _cacheInstance;
+
+    internal record struct Entry(string ClassId, GameObject Prefab);
+
+    /// <summary> Adds the given prefab to the cache. </summary>
+    /// <param name="prefab"> The prefab object that is disabled and cached. </param>
+    public static void AddPrefab(GameObject prefab)
+    {
+        EnsureCacheExists();
+    }
+
+    /// <summary>
+    /// Returns
+    /// </summary>
+    /// <param name="classId"></param>
+    /// <returns></returns>
+    public static bool IsPrefabCached(string classId)
+    {
+        if (_cacheInstance == null)
+            return false;
+        return _cacheInstance.Entries.ContainsKey(classId);
+    }
+
+    /// <summary>
+    /// Returns true if the prefab matching the given <paramref name="classId"/> is found in the cache, where it then assigns the <paramref name="prefab"/> out parameter.
+    /// </summary>
+    /// <param name="classId"></param>
+    /// <param name="prefab"></param>
+    /// <returns></returns>
+    public static bool TryGetPrefabFromCache(string classId, out GameObject prefab)
+    {
+        EnsureCacheExists();
+
+        if (_cacheInstance.Entries.TryGetValue(classId, out var found))
+        {
+            prefab = found.Prefab;
+            return prefab != null;
+        }
+
+        prefab = null;
+        return false;
+    }
+
+    private static void EnsureCacheExists()
+    {
+        if (_cacheInstance != null)
+            return;
+        _cacheInstance = new GameObject("Nautilus.PrefabCache").AddComponent<ModPrefabCacheInstance>();
+    }
+}
+internal class ModPrefabCacheInstance : MonoBehaviour
+{
+    public Dictionary<string, ModPrefabCache.Entry> Entries { get; } = new Dictionary<string, ModPrefabCache.Entry>();
```

Comment:
I don't quite understand why you're using <string, Entry> instead of simply <string, GameObject> since the Entry simply has the class id and gameobject anyway. You're double-caching the class id for seemingly no reason

For example, https://github.com/SubnauticaModding/Nautilus/pull/368/commits/e55983a70678cb976a5c89bf027d8632d79b394f#diff-68ebec32acc0b25052a6c274ed9a95d33aaa45c66ef559c335291fe69632e150R41 could be turned into simply 
```
return _cacheInstance.Entries.TryGetValue(classId, out prefab) && prefab != null;
```

Reply:
In case we ever want to extend the cache (like how we had a tuple in the past with some sort of boolean), I would say it's a lot more maintainable to have the data stored in structs than use tuples,

Reply:
However I think that tuple was incredibly useless so we can probably get away with removing it.

Reply:
I don’t disagree with the premise, but since this is all internal/private anyway, I don’t think it’s a big deal. Even if someone does need to add more to the struct, it’s not hard for them to simply change the dictionary to use the struct again. A change to one means you can change the other, they’re even in the same file.